tests/src/common/test_case.rs

```diff
@@ -288,9 +291,197 @@ impl<Dispatcher, Treasury, Profit, Leaser, Lpp, Oracle, TimeAlarms>
     }
 }
 
-pub struct TestCase<Dispatcher, Treasury, Profit, Leaser, Lpp, Oracle, TimeAlarms> {
-    pub app: MockApp,
-    pub message_receiver: WrappedCustomMessageReceiver,
+#[must_use]
+pub(crate) struct WrappedApp {
```

Comment:
May we find better names than "WrappedXyz"?

Reply:
I'll be happy to hear any suggestions, because I can't think of any right now.

Reply:
it's part of the developer's responsabilities